akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -433,14 +437,24 @@ object ShardCoordinator {
 
     def receive = {
       case BeginHandOffAck(`shard`) =>
-        remaining -= sender()
-        if (remaining.isEmpty) {
-          from ! HandOff(shard)
-          context.become(stoppingShard, discardOld = true)
-        }
+        log.debug("BeginHandOffAck for shard [{}] received from {}", shard, sender())
+        acked(sender())
```

Comment:
This means receiving a BeginHandOffAck calls an unwatch in the acked function, which makes sense for receiving Terminated but less so here.

Reply:
I think it's the other way around - unwatching is automatic when you received a `Terminated`, but in this case, if a terminating actor sends a `BeginHandOffAck`, we don't want to receive it's `Terminated` afterwards because we've already handled that actor.

Reply:
Thanks for the explanation 👍